tests/rptest/services/redpanda.py

```diff
@@ -143,8 +143,6 @@ def start(self, nodes=None, clean_nodes=True):
             assert set(node.ns) == {"redpanda"}
             assert set(node.ns["redpanda"].topics) == {"controller", "kvstore"}
 
-        self._create_initial_topics()
```

Comment:
Currently, the initially created topics will not be associated with the SASL user since the topics are made first. This would cause an error if someone tried to produce/consume to/from the topic with the SASL user. Switching up the ordering and using the security creds with topic creation fixed the problem.

All ducktape tests pass with this change on my local machine.